src/pages/AddMembersPage/useCircleTokens.ts

```diff
@@ -0,0 +1,82 @@
+import assert from 'assert';
+
+import { useQuery } from 'react-query';
+
+import { CircleTokenType } from '../../common-lib/circleShareTokens';
+import { client } from '../../lib/gql/client';
+
+export const useMagicToken = (circleId: number) => {
+  return useCircleTokens(circleId, CircleTokenType.Magic);
+};
+
+export const useWelcomeToken = (circleId: number) => {
+  return useCircleTokens(circleId, CircleTokenType.Welcome);
+};
+
+export const deleteMagicToken = (circleId: number) => {
+  return deleteToken(circleId, CircleTokenType.Magic);
+};
+
+export const deleteWelcomeToken = (circleId: number) => {
+  return deleteToken(circleId, CircleTokenType.Welcome);
+};
+
+const useCircleTokens = (circleId: number, type: CircleTokenType) => {
+  return useQuery(
+    ['circle-token-', circleId, type],
+    async (): Promise<string> => {
+      const { circle_share_tokens } = await client.query({
```

Comment:
will we need to check here if circle is already soft deleted ?

Reply:
yeah i guess, that stuff got added after i built this (this code is getting old....)

Reply:
ok i updated this